app/src/main/kotlin/com/simplemobiletools/calculator/helpers/GetCryptoTask.kt

```diff
@@ -0,0 +1,94 @@
+package com.simplemobiletools.calculator.helpers
+
+import android.os.AsyncTask
+import android.util.Log
+import com.simplemobiletools.calculator.activities.CryptoActivity
+import com.simpletools.calculator.commons.helpers.Formatter
+import org.json.JSONException
+import org.json.JSONObject
+import java.io.BufferedInputStream
+import java.io.BufferedReader
+import java.io.InputStreamReader
+import java.net.HttpURLConnection
+import java.net.URL
+
+class GetCryptoTask(val cryptoFROM: String, val cryptoTO: String, cryptoActivity: CryptoActivity, cryptoCalculatorImpl: CryptoCalculatorImpl) : AsyncTask<Void, Void, String>() {
+
+    private val cryptoActivity: CryptoActivity = cryptoActivity
+    private var cryptoCalculator: CryptoCalculatorImpl = cryptoCalculatorImpl
+
+    override fun doInBackground(vararg params: Void?): String? {
+        try {
+            val url = URL("https://api.cryptonator.com/api/ticker/$cryptoFROM-$cryptoTO")
+
+            val httpClient = url.openConnection() as HttpURLConnection
+            httpClient.setRequestProperty("Content-Type", "application/x-www-form-urlencoded")
+            httpClient.setRequestProperty("charset", "utf-8")
+            httpClient.useCaches = false
+
+            if (httpClient.responseCode == HttpURLConnection.HTTP_OK) {
+                val stream = BufferedInputStream(httpClient.inputStream)
+                val data: String = readStream(inputStream = stream)
+                httpClient.disconnect()
+                return data
+            } else if (httpClient.responseCode == HttpURLConnection.HTTP_NOT_FOUND) {
```

Comment:
@miisaka  I would delete all those else if and only keep the global else. You are basically doing the same thing in every `else if`

Reply:
I have removed them to improve code quality.